src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_text.cs

```diff
@@ -2,74 +2,114 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Buffers.Text;
+using System.Runtime.CompilerServices;
 
 namespace System.Buffers.Reader
 {
-    public static partial class BufferReaderExtensions 
+    public ref partial struct BufferReader
     {
-        public static bool TryParse(ref BufferReader reader, out bool value) 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe bool TryParse(out bool value)
         {
-            var unread = reader.UnreadSegment;
-            if (Utf8Parser.TryParse(unread, out value, out int consumed))
+            var unread = UnreadSegment;
+            if (Utf8Parser.TryParse(unread, out value, out int consumed) && consumed < unread.Length)
```

Comment:
Why do we need to compare `consumed` with `unread.Length` here? Isn't parsing bool known to be of constant length (i.e. either 4 or 5)? Shouldn't we return true if TryParse returns true, regardless?

Reply:
For bool I can see dropping out. The others don't know if there is more valid data beyond a segment. I'll change this one and add a comment.